### ISSUE TITLE
NetStandard support

### ISSUE THREAD

**Issue by shatl** [xmlunit/xmlunit.net#21]
Please add NetStandard 1.6 support

**Comment by bodewig** [comment]
I'm not opposed to it, just need a way to figure out how to do it without too much disruption. I certainly don't want to stop supporting .NET 3.5.

AFAICT NUnit 2.x doesn't support .NET Standard, so we can only provide Core and the NUnit3 Constraints.

**Comment by shatl** [comment]
XPath is there https://www.nuget.org/packages/System.Xml.XPath/

**Comment by bodewig** [comment]
Ahh, thank you, overlooked that.

Have you got any experience porting a library to .NET Core without disturbing the existing build too much?

My own experience is with Apache log4net and there we've set up a completely independent build for .NET Core that [references](https://github.com/apache/log4net/blob/1e1b82058cbe970704ff4249935a235f1c135dae/netstandard/log4net/project.json#L10) sources from the original source tree combined with quite a few pre-processor macros scatter through the code. This feels cumbersome and I'd prefer to take a simpler approach if there was one.

**Comment by shatl** [comment]
I used the same approach.

Another option is to use CSProj for .Net Core and shared projects, but I
don't think is will be simplier

Regards,
  Vitaliy

**Comment by bodewig** [comment]
If you could spare some time to help out with setting this up, it would be great. Otherwise I'll take care of it myself, but it may take a bit of time (as in weeks).

**Comment by shatl** [comment]
Looks like it's not possible to port it right now, attached it portability analyzer output. 
Main issue is lack of XML Schema support


[ApiPortAnalysis.xlsx](https://github.com/xmlunit/xmlunit.net/files/722120/ApiPortAnalysis.xlsx)

**Comment by bodewig** [comment]
Many thanks for your analysis. So if we have to wait for 2.0 I'll remove the milestone again :-)

**Comment by shatl** [comment]
@bodewig Good news , looks like it is compatible with .net standard 2.0 
[ApiPortAnalysis.xlsx](https://github.com/xmlunit/xmlunit.net/files/1356952/ApiPortAnalysis.xlsx)

**Comment by bodewig** [comment]
as long as I can use msbuild/xbuild and can still provide assemblies for .NET 3.5 I'll be fine with it.

**Comment by rmourato** [comment]
Hi @bodewig, @shatl. Did this work ever complete? Having a NET Standard compatible release would be really handy...

**Comment by bodewig** [comment]
I don't really know. I've not been working on it myself and it looks like @shatl 's code never reached the point where a PR had been opened.

**Comment by shatl** [comment]
@bodewig Please take a look at #28 .

There are several breaking changes:
* Dropped NUnit 2 support and .NET 3,5;
* AppVeyor build needs fixes;
* Nuget was not tested, pretty sure it is broken now

**Comment by bodewig** [comment]
Actually while I'm totally fine with adding .NET Standard 2 as an option I don't want to break support for .NET 3.5 or NUnit 2 at all. I was hoping for a solution that allows both to live next to each other.

**Comment by shatl** [comment]
As far as I remember .NET 3.5 support was ended around 2011. Now I'm not even sure where to get 3.5 targeting SDK for Visual Studio 2017.
I tried to convert it, but then realized 3.5 is not present on my machine.

**Comment by bodewig** [comment]
First of all I forgot to thank you @shatl, I really appreciate your help.

I don't deny .NET 3.5 is obsolete for many. The nuget package for NUnit 2.x support has more downloads than the NUnit 3.x support package, so people are still using the older platforms and I don't want to  alienate them.

An approach that I would have considered was additional .NET Stamdard build files next to the existing build. I can still give it a try based on your work, but am pretty busy with other things so it might take some more time.

**Comment by shatl** [comment]
Looks like I found the fix, hope to have some time for it this weekend.

**Comment by bodewig** [comment]
sounds great

**Comment by WeLikeAich** [comment]
Hi I was wondering if there was any update on this issue. I too would love a .net standard 2.1/2.2 version of this library as it does everything I need.

Thanks!

**Comment by bodewig** [comment]
So far I believe #28 is a work in progress and I'm waiting for it to become ready.

**Comment by shatl** [comment]
@bodewig @WeLikeAich Finally I found time for this :)

Conflicts resolved, please review

**Comment by lucastheisen** [comment]
@bodewig , any chance this is going to get resolved soon?

**Comment by bodewig** [comment]
Somehow I managed to miss @shatl 's comment from mid-February, sorry. There are a few tweaks I'd like to see and I hope to find the time to perform them the coming days.

Thanks for your patience.

**Comment by bodewig** [comment]
see the netstandard-2.0 for the work in progress

**Comment by bodewig** [comment]
@lucastheisen if you could try building the .NET Standard part of the netstandard2.0 branch, that would be really helpful. I think it is in a state now where I could build beta-packages and publish them.

**Comment by lucastheisen** [comment]
@bodewig , It builds (with 22 warnings), and all three test projects run without error (some tests skipped):
```
PS C:\Users\ltheisen\git\xmlunit-xmlunit.net> dotnet build .\XMLUnit.NET.sln
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj...
  Restoring packages for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-placeholders\XMLUnit.Placeholders.csproj...
  Restoring packages for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\XMLUnit.Placeholders.Tests.csproj...
  Restoring packages for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-core\XMLUnit.Core.Tests.csproj...
  Restoring packages for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.csproj...
  Restoring packages for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj...
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\obj\XMLUnit.Core.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-placeholders\obj\XMLUnit.Placeholders.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\obj\XMLUnit.Core.csproj.nuget.g.targets.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-placeholders\obj\XMLUnit.Placeholders.csproj.nuget.g.targets.
  Restore completed in 286.39 ms for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj.
  Restore completed in 286.39 ms for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-placeholders\XMLUnit.Placeholders.csproj.
  Installing NUnit 3.6.1.
  Installing System.ComponentModel.EventBasedAsync 4.3.0.
  Installing NUnit3TestAdapter 3.10.0.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\obj\XMLUnit.NUnit3.Constraints.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\obj\XMLUnit.NUnit3.Constraints.csproj.nuget.g.targets.
  Restore completed in 1.59 sec for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.csproj.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\obj\XMLUnit.Placeholders.Tests.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\obj\XMLUnit.NUnit3.Constraints.Test.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-core\obj\XMLUnit.Core.Tests.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\obj\XMLUnit.Placeholders.Tests.csproj.nuget.g.targets.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-core\obj\XMLUnit.Core.Tests.csproj.nuget.g.targets.
  Generating MSBuild file C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\obj\XMLUnit.NUnit3.Constraints.Test.csproj.nuget.g.targets.
  Restore completed in 4.07 sec for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\XMLUnit.Placeholders.Tests.csproj.
  Restore completed in 4.07 sec for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj.
  Restore completed in 4.07 sec for C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-core\XMLUnit.Core.Tests.csproj.
Util\Convert.cs(62,13): warning CS0618: 'XmlReaderSettings.ProhibitDtd' is obsolete: 'Use XmlReaderSettings.DtdProcessing property instead.' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Validation\Validator.cs(133,43): warning CS0618: 'ValidationType.XDR' is obsolete: 'XDR Validation through XmlValidatingReader is obsoleted' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(31,26): warning CS1572: XML comment has a param tag for 'comparison', but there is no parameter by that name [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(47,49): warning CS1573: Parameter 'difference' has no matching param tag in the XML comment for 'DefaultComparisonFormatter.GetDescription(Comparison)' (but other parameters do) [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(172,94): warning CS1573: Parameter 'dec' has no matching param tag in the XML comment for 'DefaultComparisonFormatter.AppendDocumentXmlDeclaration(StringBuilder, XmlDeclaration)' (but other parameters do) [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(315,26): warning CS1572: XML comment has a param tag for 'details', but there is no parameter by that name [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(328,52): warning CS1573: Parameter 'difference' has no matching param tag in the XML comment for 'DefaultComparisonFormatter.GetDetails(Comparison.Detail, ComparisonType, bool)' (but other parameters do) [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
  XMLUnit.Core -> C:\Users\ltheisen\git\xmlunit-xmlunit.net\build\bin\Debug\netstandard2.0\xmlunit-core.dll
PlaceholderDifferenceEvaluator.cs(58,39): warning CS1591: Missing XML comment for publicly visible type or member 'PlaceholderDifferenceEvaluator.PLACEHOLDER_DEFAULT_OPENING_DELIMITER_REGEX' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-placeholders\XMLUnit.Placeholders.csproj]
PlaceholderDifferenceEvaluator.cs(59,39): warning CS1591: Missing XML comment for publicly visible type or member 'PlaceholderDifferenceEvaluator.PLACEHOLDER_DEFAULT_CLOSING_DELIMITER_REGEX' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-placeholders\XMLUnit.Placeholders.csproj]
EvaluateXPathConstraint.cs(44,1): warning CS1570: XML comment has badly formed XML -- 'Expected an end tag for element 'code'.' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.csproj]
EvaluateXPathConstraint.cs(44,1): warning CS1570: XML comment has badly formed XML -- 'Expected an end tag for element 'example'.' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.csproj]
EvaluateXPathConstraint.cs(47,9): warning CS1570: XML comment has badly formed XML -- 'End tag was not expected at this location.' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.csproj]
  XMLUnit.Placeholders -> C:\Users\ltheisen\git\xmlunit-xmlunit.net\build\bin\Debug\netstandard2.0\xmlunit-placeholders.dll
  XMLUnit.NUnit3.Constraints -> C:\Users\ltheisen\git\xmlunit-xmlunit.net\build\bin\Debug\netstandard2.0\xmlunit-nunit3-constraints.dll
C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\shared\CommonAssemblyInfo.cs(20,28): warning CS0436: The type 'XmlUnitVersion' in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\..\..\shared\CommonAssemblyInfo.cs' conflicts with the imported type 'XmlUnitVersion' in 'xmlunit-placeholders, Version=2.7.0.133, Culture=neutral, PublicKeyToken=e7d7e3088fd452f6'. Using the type defined in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\..\..\shared\CommonAssemblyInfo.cs'. [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\XMLUnit.Placeholders.Tests.csproj]
C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\shared\CommonAssemblyInfo.cs(21,41): warning CS0436: The type 'XmlUnitVersion' in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\..\..\shared\CommonAssemblyInfo.cs' conflicts with the imported type 'XmlUnitVersion' in 'xmlunit-placeholders, Version=2.7.0.133, Culture=neutral, PublicKeyToken=e7d7e3088fd452f6'. Using the type defined in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\..\..\shared\CommonAssemblyInfo.cs'. [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\XMLUnit.Placeholders.Tests.csproj]
C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\shared\CommonAssemblyInfo.cs(20,28): warning CS0436: The type 'XmlUnitVersion' in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\..\..\shared\CommonAssemblyInfo.cs' conflicts with the imported type 'XmlUnitVersion' in 'xmlunit-nunit3-constraints, Version=2.7.0.133, Culture=neutral, PublicKeyToken=e7d7e3088fd452f6'. Using the type defined in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\..\..\shared\CommonAssemblyInfo.cs'. [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\shared\CommonAssemblyInfo.cs(21,41): warning CS0436: The type 'XmlUnitVersion' in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\..\..\shared\CommonAssemblyInfo.cs' conflicts with the imported type 'XmlUnitVersion' in 'xmlunit-nunit3-constraints, Version=2.7.0.133, Culture=neutral, PublicKeyToken=e7d7e3088fd452f6'. Using the type defined in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\..\..\shared\CommonAssemblyInfo.cs'. [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
HasXPathConstraintTest.cs(127,44): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
EvaluateXPathConstraintTest.cs(101,44): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
HasXPathConstraintTest.cs(135,30): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
EvaluateXPathConstraintTest.cs(115,30): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
CompareConstraintTest.cs(205,44): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
CompareConstraintTest.cs(213,30): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
  XMLUnit.Core.Tests -> C:\Users\ltheisen\git\xmlunit-xmlunit.net\build\bin\Debug\netcoreapp2.0\xmlunit-core.tests.dll
  XMLUnit.Placeholders.Tests -> C:\Users\ltheisen\git\xmlunit-xmlunit.net\build\bin\Debug\netcoreapp2.0\xmlunit-placeholders.tests.dll
  XMLUnit.NUnit3.Constraints.Test -> C:\Users\ltheisen\git\xmlunit-xmlunit.net\build\bin\Debug\netcoreapp2.0\xmlunit-nunit3-constraints.tests.dll

Build succeeded.

Util\Convert.cs(62,13): warning CS0618: 'XmlReaderSettings.ProhibitDtd' is obsolete: 'Use XmlReaderSettings.DtdProcessing property instead.' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Validation\Validator.cs(133,43): warning CS0618: 'ValidationType.XDR' is obsolete: 'XDR Validation through XmlValidatingReader is obsoleted' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(31,26): warning CS1572: XML comment has a param tag for 'comparison', but there is no parameter by that name [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(47,49): warning CS1573: Parameter 'difference' has no matching param tag in the XML comment for 'DefaultComparisonFormatter.GetDescription(Comparison)' (but other parameters do) [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(172,94): warning CS1573: Parameter 'dec' has no matching param tag in the XML comment for 'DefaultComparisonFormatter.AppendDocumentXmlDeclaration(StringBuilder, XmlDeclaration)' (but other parameters do) [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(315,26): warning CS1572: XML comment has a param tag for 'details', but there is no parameter by that name [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
Diff\DefaultComparisonFormatter.cs(328,52): warning CS1573: Parameter 'difference' has no matching param tag in the XML comment for 'DefaultComparisonFormatter.GetDetails(Comparison.Detail, ComparisonType, bool)' (but other parameters do) [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-core\XMLUnit.Core.csproj]
PlaceholderDifferenceEvaluator.cs(58,39): warning CS1591: Missing XML comment for publicly visible type or member 'PlaceholderDifferenceEvaluator.PLACEHOLDER_DEFAULT_OPENING_DELIMITER_REGEX' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-placeholders\XMLUnit.Placeholders.csproj]
PlaceholderDifferenceEvaluator.cs(59,39): warning CS1591: Missing XML comment for publicly visible type or member 'PlaceholderDifferenceEvaluator.PLACEHOLDER_DEFAULT_CLOSING_DELIMITER_REGEX' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-placeholders\XMLUnit.Placeholders.csproj]
EvaluateXPathConstraint.cs(44,1): warning CS1570: XML comment has badly formed XML -- 'Expected an end tag for element 'code'.' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.csproj]
EvaluateXPathConstraint.cs(44,1): warning CS1570: XML comment has badly formed XML -- 'Expected an end tag for element 'example'.' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.csproj]
EvaluateXPathConstraint.cs(47,9): warning CS1570: XML comment has badly formed XML -- 'End tag was not expected at this location.' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\main\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.csproj]
C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\shared\CommonAssemblyInfo.cs(20,28): warning CS0436: The type 'XmlUnitVersion' in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\..\..\shared\CommonAssemblyInfo.cs' conflicts with the imported type 'XmlUnitVersion' in 'xmlunit-placeholders, Version=2.7.0.133, Culture=neutral, PublicKeyToken=e7d7e3088fd452f6'. Using the type defined in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\..\..\shared\CommonAssemblyInfo.cs'. [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\XMLUnit.Placeholders.Tests.csproj]
C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\shared\CommonAssemblyInfo.cs(21,41): warning CS0436: The type 'XmlUnitVersion' in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\..\..\shared\CommonAssemblyInfo.cs' conflicts with the imported type 'XmlUnitVersion' in 'xmlunit-placeholders, Version=2.7.0.133, Culture=neutral, PublicKeyToken=e7d7e3088fd452f6'. Using the type defined in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\..\..\shared\CommonAssemblyInfo.cs'. [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-placeholders\XMLUnit.Placeholders.Tests.csproj]
C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\shared\CommonAssemblyInfo.cs(20,28): warning CS0436: The type 'XmlUnitVersion' in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\..\..\shared\CommonAssemblyInfo.cs' conflicts with the imported type 'XmlUnitVersion' in 'xmlunit-nunit3-constraints, Version=2.7.0.133, Culture=neutral, PublicKeyToken=e7d7e3088fd452f6'. Using the type defined in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\..\..\shared\CommonAssemblyInfo.cs'. [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\shared\CommonAssemblyInfo.cs(21,41): warning CS0436: The type 'XmlUnitVersion' in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\..\..\shared\CommonAssemblyInfo.cs' conflicts with the imported type 'XmlUnitVersion' in 'xmlunit-nunit3-constraints, Version=2.7.0.133, Culture=neutral, PublicKeyToken=e7d7e3088fd452f6'. Using the type defined in 'C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\..\..\shared\CommonAssemblyInfo.cs'. [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
HasXPathConstraintTest.cs(127,44): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
EvaluateXPathConstraintTest.cs(101,44): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
HasXPathConstraintTest.cs(135,30): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
EvaluateXPathConstraintTest.cs(115,30): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
CompareConstraintTest.cs(205,44): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
CompareConstraintTest.cs(213,30): warning CS0618: 'Is.StringContaining(string)' is obsolete: 'Deprecated, use Does.Contain' [C:\Users\ltheisen\git\xmlunit-xmlunit.net\src\tests\net-constraints-nunit3\XMLUnit.NUnit3.Constraints.Test.csproj]
    22 Warning(s)
    0 Error(s)

Time Elapsed 00:00:14.36
PS C:\Users\ltheisen\git\xmlunit-xmlunit.net> dotnet test src/tests/net-core/XMLUnit.Core.Tests.csproj
Build started, please wait...
Build completed.

Test run for C:\Users\ltheisen\git\xmlunit-xmlunit.net\build\bin\Debug\netcoreapp2.0\xmlunit-core.tests.dll(.NETCoreApp,Version=v2.0)
Microsoft (R) Test Execution Command Line Tool Version 15.9.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
Skipped  TestDiff_withoutIgnoreElementContentWhitespaces_shouldFail
Skipped  IgnorePrologExceptDoctypeDoesntIgnoreNameOfDoctype
Skipped  IgnorePrologExceptDoctypeDoesntIgnorePresenceOfDoctype
Skipped  IgnorePrologIgnoresNameOfDoctype
Skipped  IgnorePrologIgnoresPresenceOfDoctype

Total tests: 314. Passed: 309. Failed: 0. Skipped: 5.
Test Run Successful.
Test execution time: 2.2558 Seconds
PS C:\Users\ltheisen\git\xmlunit-xmlunit.net> dotnet test src/tests/net-constraints-nunit3/XMLUnit.NUnit3.Constraints.Test.csproj
Build started, please wait...
Build completed.

Test run for C:\Users\ltheisen\git\xmlunit-xmlunit.net\build\bin\Debug\netcoreapp2.0\xmlunit-nunit3-constraints.tests.dll(.NETCoreApp,Version=v2.0)
Microsoft (R) Test Execution Command Line Tool Version 15.9.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
Skipped  ShouldSuccessfullyValidateInstanceWithoutExplicitSchemaSource

Total tests: 53. Passed: 52. Failed: 0. Skipped: 1.
Test Run Successful.
Test execution time: 1.6781 Seconds
PS C:\Users\ltheisen\git\xmlunit-xmlunit.net> dotnet test src/tests/net-placeholders/XMLUnit.Placeholders.Tests.csproj
Build started, please wait...
Build completed.

Test run for C:\Users\ltheisen\git\xmlunit-xmlunit.net\build\bin\Debug\netcoreapp2.0\xmlunit-placeholders.tests.dll(.NETCoreApp,Version=v2.0)
Microsoft (R) Test Execution Command Line Tool Version 15.9.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

Total tests: 25. Passed: 25. Failed: 0. Skipped: 0.
Test Run Successful.
Test execution time: 1.5241 Seconds
```

**Comment by bodewig** [comment]
Thank you, I'll merge the branch and cut a beta release soonish.

**Comment by bodewig** [comment]
https://www.nuget.org/packages/XMLUnit.Core/2.7.0-beta-01 - release will be announced soon.

**Comment by bodewig** [comment]
Has anybody already given the beta nupkgs a try? If they work as expected, I'll be happy to create a "final" 2.7.0 release. Ideally I'd hear from .NET Standard and from Framework projects for versions prior to whatever implemented .NET Standard 2.0.

**Comment by lucastheisen** [comment]
Prior to 2.7.0-beta-01 I get these build warnings:
```
C:\Users\ltheisen\git\asiasetl-tableau-automation\Mitre.Tableau.Test\Mitre.Tableau.Test.csproj : warning NU1701: Package 'XMLUnit.Core 2.6.0' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project. [C:\Users\ltheisen\git\asiasetl-tableau-automation\Mitre.Tableau.sln]
C:\Users\ltheisen\git\asiasetl-tableau-automation\Mitre.Tableau.Test\Mitre.Tableau.Test.csproj : warning NU1701: Package 'XMLUnit.Core 2.6.0' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.
    2 Warning(s)
    0 Error(s)
```
After upgrading, they go away.  All tests continue to work as expected.

**Comment by bodewig** [comment]
Thank you

**Comment by bodewig** [comment]
just published the final 2.7.0 release. Many thanks @shatl  and @lucastheisen .
